integration-tests/suites/base.go

```diff
@@ -0,0 +1,416 @@
+package suites
```

Comment:
I would argue that this file should actually live under `suites/common/`, since most functions and methods seem to be quite generic. Alternatively, it might be worth it at some point to go over individual functions and split them into more specific files, just by going through the file I see potential splits into:
- Stats container
- Performance analysis
- Container handling
- Various assertions & DB access

This can be done in follow up though, no point on stop merging this PR on a nit.

Reply:
Absolutely agree that a lot of this can and should be split out into their own structures, and I think the base structure could then pull some of those things together to keep it all nicely separated